### PR TITLE
Feature/issues/68

### PR DESCRIPTION
**Pull request by Mizzick**

### CODE REVIEW

File: Extension/lib/utils/service-client.js [hunk 1]
```diff
@@ -38,14 +38,15 @@ var ServiceClient = exports.ServiceClient = function () {
 
 	// Base url of our backend server
 	this.backendUrl = "https://chrome.adtidy.org";
+	this.mobileBackendUrl = "http://mobile.adtidy.org/api/1.0";
```
Comment (ameshkov) [inline Extension/lib/utils/service-client.js]: TODO: Add mobile filters to extension backend instead.

File: Extension/browser/firefox/lib/prefs.js [hunk 2]
```diff
@@ -36,6 +36,10 @@ var Prefs = exports.Prefs = {
 		var url = "filters/filter_" + filterId + ".txt";
 		return self.data.url(url);
 	},
+	getLocalMobileFilterPath: function (filterId) {
```
Comment (ameshkov) [inline Extension/browser/firefox/lib/prefs.js]: I don't think that it's a right solution to embed mobile filters into the add-on.

Just work as if these filters were not embedded at all, load them from the backend.

File: Extension/browser/firefox/lib/prefs.js [hunk 3]
```diff
@@ -36,6 +36,10 @@ var Prefs = exports.Prefs = {
...
+		var url = "filters_mobile/filter_" + filterId + ".txt";
```
Comment (ameshkov) [inline Extension/browser/firefox/lib/prefs.js]: At least leave them in the same folder, just change naming to "filter_mobile_" + filterId

**Comment by ameshkov** [comment]
One more important note. Let's change the file name of the download filter too.

I mean if "optimized" flag is used, let it be **filter_mobile_123.txt**.
Thus when you switch the flag back to "non-optimized" and reset filter versions, you will be sure that the old filter is not used.

File: Compiler/src/main/java/com/adguard/compiler/FilterUtils.java [hunk 4]
```diff
@@ -85,11 +87,12 @@ public static void updateLocalFilters(File source) throws IOException {
      * It contains big javascript rule for youtube ad blocking.
      *
      * @param destDir Destination directory.
+     * @param filterDownloadUrl
      * @throws IOException
      */
-    public static void loadEnglishFilterForSafari(File destDir) throws IOException {
+    public static void loadEnglishFilterForSafari(File destDir, String filterDownloadUrl) throws IOException {
```
Comment (ameshkov) [inline Compiler/src/main/java/com/adguard/compiler/FilterUtils.java]: In fact we don't need this method anymore as we now have safari filter.

File: Compiler/src/main/java/com/adguard/compiler/FilterUtils.java [hunk 5]
```diff
@@ -100,11 +103,12 @@ public static void loadEnglishFilterForSafari(File destDir) throws IOException {
      * This filters contains some special fix rules for safari content blocker
      *
      * @param destDir Destination directory.
+     * @param filterDownloadUrl
      * @throws IOException
      */
-    public static void loadMobileSafariFilter(File destDir) throws IOException {
+    public static void loadMobileSafariFilter(File destDir, String filterDownloadUrl) throws IOException {
```
Comment (ameshkov) [inline Compiler/src/main/java/com/adguard/compiler/FilterUtils.java]: We don't need this method, safari filter is downloaded as ordinary local filter.

**Comment by atropnikov** [comment]
We need to change logic of call method 'reloadAntiBannerFilters'. Now you call it from options page, you need subscribe to event CHANGE_USER_SETTINGS with type USE_OPTIMIZED_FILTERS.

**Comment by Mizzick** [comment]
>We need to change logic of call method 'reloadAntiBannerFilters'. Now you call it from options page, you need subscribe to event CHANGE_USER_SETTINGS with type USE_OPTIMIZED_FILTERS.

This is not possible, I can't call antiBannerService.reloadAntiBannerFilters() from options page.

**Comment by atropnikov** [comment]
You don't need call reloadAntiBannerFilters from options page. Subscribe to event in antibannerService.

**Comment by Mizzick** [comment]
Fixed

**Pull request merged**